python/cudnn/sdpa/fwd/api_sm80.py

```
@@ -0,0 +1,818 @@
+            # "causal_swa" (MASK_CAUSAL|MASK_SWA — causal upper bound kept).
+            # window_size_right > 0 widens the causal upper bound into a right
+            # band (k <= q + right) via the kernel's runtime right_bound.
+            self.mask_token = "causal" if swa_left < 0 else "causal_swa"
+            self.swa_window_runtime = max(0, swa_left) if swa_left >= 0 else 0
+            self.right_bound = max(0, swa_right)
+        elif swa_left >= 0 or swa_right >= 0:
+            self._not_implemented_error_if(
+                swa_right > 0,
+                "SM80 SDPA: non-causal SWA with window_size_right > 0 is not supported " "(window_size_right is only meaningful with is_causal=True)",
+            )
+            self.mask_token = "swa"
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "mask_token"

(S105)